### PR TITLE
HIP - associate/dissociate 

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
HIP proposal to add `associate` and `dissociate` functionality to the proxy facade contract for tokens.

### CODE REVIEW

File: hip-associate-dissociate.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+[HIP-218](https://hips.hedera.com/hip/hip-218) proposed a way to enable treating HTS tokens in an equivalent manner
+to ERC-20 and ERC-721 tokens by creating a proxy redirect facade contract.  This proposal extends
+this functionality to include the HTS `associate` and `dissociate` functions.
+
+## Motivation
+
+There are some functions that are unique to Hedera HTS tokens that are not part of the ERC standard.
+A mechanism should be provided to enable developers to call these functions on HTS tokens in a way that is familiar
+to users of ERC-20 and ERC-721 tokens.
```
Comment (Nana-EC) [inline hip-associate-dissociate.md:28]: Additionally direct user calling support for EOAs

File: hip-associate-dissociate.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+## Rationale
+
+By adding `associate` and `dissociate` functions to the token proxy redirect facade contract, functionality
+that is only available to HTS tokens can be included to the available capabilities of token contracts.
+
+## User stories
+
+1. As a developer, I want to be able to call the `associate` and `dissociate` functions on HTS tokens in a way that is similar to calling functions on the IERC20 and IERC721 interfaces.
```
Comment (Nana-EC) [inline hip-associate-dissociate.md:37]: As a DApp I want to support a users ability to execute the association flows without having to call an intermediate contract.
Something along that line

Comment (Ashe-Oro) [reply hip-associate-dissociate.md]: and it's really great UX to be able to include the token associate in the same contract call as a claim function, altho not sure if that's gonna be possible in any way under the new security model. Having to associate in an tx and then claim in another tx is meh UX. 

Granted, the same thing could be said about requiring spending approval in a separate tx (just pointing out some potential hypocrisy on my side). Guess I see the associate action is just so much lower risk than a spending approval, so perhaps it doesn't require a separate tx.

File: HIP/hip-associate-dissociate.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+1. As a developer, I want to be able to call the `associate` and `dissociate` functions on HTS tokens in a way that is similar to calling functions on the IERC20 and IERC721 interfaces.
+2. As a DApp, I want to be able to enable users to call the `associate` and `dissociate` functions without having to have intermediary contracts.
+
+## Specification
+
+The method by which the redirection to proxy contract is currently implemented is that during the execution of an EVM request, 
+the contract address is examined to determine whether it is an HTS token address and if so then the request execution 
+is redirected to the proxy contract.
+
+More specifically the algorithm by which the system get from (a) the call to the token address to (b) the precompiled contract is as follows:
```
Comment (Nana-EC) [inline HIP/hip-associate-dissociate.md:47]:
```suggestion
More specifically the algorithm by which the system gets from (a) the call to the token address to (b) the precompiled contract is as follows:
```

**Review by Nana-EC — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]

**Review by mgarbs — APPROVED** [review]

**Pull request merged**